plugins/modules/aci_route_control_match_dest.py

```diff
@@ -0,0 +1,326 @@
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        subject_name=dict(type="str"),
+        ip=dict(type="str"),
+        aggregate=dict(type="bool"),
+        greater_than_mask=dict(type="int", aliases=["greater_than", "gt"]),
+        less_than_mask=dict(type="int", aliases=["less_than", "lt"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "subject_name", "ip"]],
+            ["state", "present", ["tenant", "subject_name", "ip"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    subject_name = module.params.get("subject_name")
+    ip = module.params.get("ip")
+    aggregate = aci.boolean(module.params.get("aggregate"))
+    greater_than = module.params.get("greater_than_mask")
+    less_than = module.params.get("less_than_mask")
+    state = module.params.get("state")
+
+    ipv4_regex = r"^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/(?:[0-9]|[0-2][0-9]|3[0-2])$"
+    ipv6_regex = r"^(?:(?:(?:[A-Fa-f0-9]{1,4}::?){1,7}[A-Fa-f0-9]{1,4})|::[A-Fa-f0-9]{1,4})/(?:[0-9]|[0-9][0-9]|1[01][0-9]|12[0-8])$"
+    combined_regex = r"(?:{0}|{1})".format(ipv4_regex, ipv6_regex)
+
+    if ip is not None:
+        if "/" not in ip:
+            aci.fail_json("ip must include the prefix length, e.g. '10.20.30.0/24' or 'fd80::/64'")
+        else:
+            mask = int(ip.split("/")[1])
+            if not re.search(combined_regex, ip):
+                aci.fail_json("ip must be a valid IPv4 or IPv6 prefix, e.g. '10.20.30.0/24' or 'fd80::/64'")
```

Comment:
Would an if/elif/else statement be an option?
```suggestion
        if "/" not in ip:
            aci.fail_json("ip must include the prefix length, e.g. '10.20.30.0/24' or 'fd80::/64'")
        elif not re.search(combined_regex, ip):
            aci.fail_json("ip must be a valid IPv4 or IPv6 prefix, e.g. '10.20.30.0/24' or 'fd80::/64'")
        else:
            mask = int(ip.split("/")[1])
```